### DISCUSSION TITLE
JSVM equivalent of tokens.NewUserAuthToken(app, user) ?

### DISCUSSION THREAD

**Discussion by madebyjono** [pocketbase/pocketbase#5441] [Q&A]
I'm trying to implement my own SMS OTP authentication flow with Pocketbase with Javascript. 

Looking at this discussion #279, I'm trying to generate the user's auth token programmatically, but in the reference JVSM docs I can't find a userAuthToken / equivalent. 

Can someone point me in the right direction? 

Thanks! :)

**Comment by ganigeorgiev** [comment]
The linked issue is very old and I don't recommend following it (it is for PocketBase < 0.8 and a lot of things has changed since then, including how users are managed).

In general all JSVM tokens helpers could be found under the global [`$tokens.*`](https://pocketbase.io/jsvm/modules/_tokens.html).

An example community provided OTP implementation via the hooks could be found in https://github.com/pocketbase/pocketbase/issues/119#issuecomment-1881906233.

Note that OTP (with email code by default, but users will be able to intercept the email action and instead send an SMS) is implemented already as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and will be available with the upcoming v0.23.0 release soon (_there are no ETAs yet_).

**Comment by madebyjono** [comment]
Thank you @ganigeorgiev for the tip of #119 - got an OTP auth going :) 

Meanwhile, excited and appreciative of the upcoming refactoring! Cheers!
